fitlins/interfaces/afni.py

```diff
@@ -80,16 +84,62 @@ def _run_interface(self, runtime):
         afni_design = get_afni_design_matrix(mat, contrasts, stim_labels, t_r)
         Path(design_fname).write_text(afni_design)
 
-        # Get input scans for 3dREMLfit: scaled and/or smoothed input and the mask file
-        img_path = self.inputs.bold_file
-        img = nb.load(img_path)
-
         mask_file = self.inputs.mask_file
         if not isdefined(mask_file):
             mask_file = None
+
+        # Need to do smoothing before scaling
         smoothing_fwhm = self.inputs.smoothing_fwhm
+        smoothing_type = self.inputs.smoothing_type
         if not isdefined(smoothing_fwhm):
             smoothing_fwhm = None
+            # Get input scans for 3dREMLfit
+            # If there's not smoothing, then the input bold file will do
+            img_path = self.inputs.bold_file
+            img = nb.load(img_path)
+        elif isdefined(smoothing_type) and smoothing_type == 'isoblurto':
```

Comment:
This seems odd to do internal to the node. Is there a reason that we can't create a new node that does the smoothing in the workflow?

Reply:
Do you mean having a smoothing node prior to either estimator, or do you mean having the afni interface be encapsulated in a workflow with distinct smoothing and estimator nodes?

Reply:
Just a node prior to the estimator. It should be skippable if the estimator will do the smoothing internally, e.g. with nistats and isometric smoothing.